lua/lualine/components/filename.lua

```diff
@@ -43,6 +43,10 @@ M.update_status = function(self)
   elseif self.options.path == 2 then
     -- absolute path
     data = vim.fn.expand('%:p')
+  elseif self.options.path == 3 then
+    -- absolute path, with tilde
+    local home_dir = os.getenv('HOME')
+    data = '~/' .. string.sub(vim.fn.expand('%:p'), #home_dir + 2)
```

Comment:
This doesn't make sense. What happens when your home is /home/asdf and you've opened a file in /usr/share . Use builtin filename modifications with expand instead (something like this `expend('%:p:~)`).